src/test/java/tigase/server/xmppclient/StreamManagementIOProcessorTest.java

```diff
@@ -197,4 +199,37 @@ public void testForceDelayForResumption() {
 				.forEach((OutQueue.Entry e) -> assertFalse(
 						e.getPacketWithStamp().isXMLNSStaticStr(new String[]{"iq", "delay"}, DELAY_XMLNS)));
 	}
+
+	@Test
+	public void testShouldRequestAck()
+			throws NoSuchFieldException, IllegalAccessException, TigaseStringprepException, InterruptedException {
+		StreamManagementIOProcessor processor = new StreamManagementIOProcessor();
+		Field ackRefMinDelayField = StreamManagementIOProcessor.class.getDeclaredField("ack_request_min_delay");
+		ackRefMinDelayField.setAccessible(true);
+		ackRefMinDelayField.set(processor, 20l);
+
+		OutQueue outQueue = new OutQueue();
+
+		assertFalse(processor.shouldRequestAck(null, outQueue));
+		for (int i = 0; i < 10; i++) {
+			Packet p = Packet.packetInstance(
+					new Element("message").withElement("body", null, "Test " + UUID.randomUUID().toString()));
+			outQueue.append(p, 2000, 2000);
+			assertEquals(i == 9, processor.shouldRequestAck(null, outQueue));
+		}
+		outQueue.sendingRequest();
+		assertFalse(processor.shouldRequestAck(null, outQueue));
+		for (int i = 0; i < 10; i++) {
+			Packet p = Packet.packetInstance(
+					new Element("message").withElement("body", null, "Test " + UUID.randomUUID().toString()));
+			outQueue.append(p, 2000, 2000);
+			assertFalse(processor.shouldRequestAck(null, outQueue));
```

Comment:
Nitpick: maybe add a comment here that we are testing second case (i.e. reducing sending `<r/>` for batch processing)? One can figure it out (especially if it's fresh reviewing code and reading intention of the change) but I prefer to have things explicit - makes it easier in the future.

Reply:
You write once, you read often - lowering amount of things you have to go through by adding simple comments would be beneficial IMHO, but as I said: nitpick.